js/bitmex.js

```diff
@@ -217,17 +219,179 @@ module.exports = class bitmex extends Exchange {
+                    'ETH': 'eth',
+                    'TRC20': 'tron',
+                    'TRX': 'tron',
+                    'BEP20': 'bsc',
+                    'SOLANA': 'sol',
+                    'AVALANCHEC': 'avax',
+                    'NEAR': 'near',
+                    'TEZOS': 'xtz',
+                    'POLKADOT': 'dot',
+                    'CARDANO': 'ada',
+                },
+                'networksById': {
+                    'btc': 'BTC',
+                    'eth': 'ERC20',
+                    'tron': 'TRC20',
+                    'bsc': 'BEP20',
+                    'sol': 'SOLANA',
+                    'avax': 'AVALANCHEC',
+                    'near': 'NEAR',
+                    'xtz': 'TEZOS',
+                    'dot': 'POLKADOT',
+                    'ada': 'CARDANO',
+                },
             },
             'commonCurrencies': {
-                'USDt': 'USDT',
-                'XBt': 'BTC',
                 'XBT': 'BTC',
-                'Gwei': 'ETH',
                 'GWEI': 'ETH',
             },
         });
     }
 
+    async fetchCurrencies (params = {}) {
+        /**
+         * @method
+         * @name bitmex#fetchCurrencies
+         * @description fetches all available currencies on an exchange
+         * @param {object} params extra parameters specific to the mexc3 api endpoint
+         * @returns {object} an associative dictionary of currencies
+         */
+        const response = await this.publicGetWalletAssets (params);
+        //
+        //    {
+        //        "XBt": {
+        //            "asset": "XBT",
+        //            "currency": "XBt",
+        //            "majorCurrency": "XBT",
+        //            "name": "Bitcoin",
+        //            "currencyType": "Crypto",
+        //            "scale": "8",
+        //            // "mediumPrecision": "8",
+        //            // "shorterPrecision": "4",
+        //            // "symbol": "₿",
+        //            // "weight": "1",
+        //            // "tickLog": "0",
+        //            "enabled": true,
+        //            "isMarginCurrency": true,
+        //            "minDepositAmount": "10000",
+        //            "minWithdrawalAmount": "1000",
+        //            "maxWithdrawalAmount": "100000000000000",
+        //            "networks": [
+        //                {
+        //                    "asset": "btc",
+        //                    "tokenAddress": "",
+        //                    "depositEnabled": true,
+        //                    "withdrawalEnabled": true,
+        //                    "withdrawalFee": "20000",
+        //                    "minFee": "20000",
+        //                    "maxFee": "10000000"
+        //                }
+        //            ]
+        //        },
+        //     }
+        //
+        const result = {};
+        this.options['currencyIdsByAssetNames'] = {};
+        const keys = Object.keys (response);
```

Comment:
@ttodua huh `response` is an array, so `Object.keys()` will be transpiled to `.keys()` in python and fail because `lists` do not have this method.
Please after each change make some quick tests on python, it's very important
![image](https://user-images.githubusercontent.com/43336371/208093821-e95bf7c0-464d-4676-adda-c0cf405bc0cd.png)

Reply:
fixed